comp/dogstatsd/server/parse.go

```diff
@@ -232,6 +224,30 @@ func (p *parser) parseMetricSample(message []byte) (dogstatsdMetricSample, error
 	}, nil
 }
 
+// parseLocalData parses the local data string into a LocalData struct.
+func (p *parser) parseLocalData(rawLocalData []byte) origindetection.LocalData {
+	localDataString := string(rawLocalData)
+
+	localData, err := origindetection.ParseLocalData(localDataString)
+	if err != nil {
+		log.Errorf("failed to parse c: field containing Local Data %q: %v", localDataString, err)
+	}
+
+	return localData
+}
+
+// parseExternalData parses the local data string into a LocalData struct.
```

Comment:
`local data string into a LocalData struct`  - should this say external data?

Reply:
Yes!